### DISCUSSION TITLE
Frequently moving Kinematic bodies not colliding with Dynamic bodies

### DISCUSSION THREAD

**Discussion by robinlacey** [bepu/bepuphysics2#330] [Q&A] [answered]
Hello

I have a number of planets that orbit a sun. The position is set each frame by updating Pose.Position. The player (a Dynamic body) is attracted towards planets by applying a small velocity on each frame. 

The problem is, if the planet's orbit is going toward the player body it will not collide with the planet. However collisions work fine when the planet is moving away from the dynamic object. 

I'm assuming that the Pose.Position change is teleporting the planet over the player dyanmic object and causing issues. 

What's the best way to do this? Is there a way to force a collision if/when player goes into planet?

Many thanks 

PS: A lot of of the other game logic is based around knowing where the planet will be at any frame (`GetPlanetPosition(atFrame)`) so im cautious about using velocity to set positions as they may not be exact?

**Comment by RossNordby** [comment, marked as answer]
That sounds like it is indeed because of teleporting the body. Setting the velocity should resolve the collision issues.

While tiny numerical error will arise, it won't accumulate over many frames if the velocity is set by simply using `(targetPosition - currentPosition) / dt`, where `targetPosition` is where you want the body to be after the timestep and `dt` is the timestep duration.

If desired, you could also simply set the position after the end of the timestep so that the position is bitwise identical to your target.

**Reply by robinlacey** [reply]
That works pefectly, thank you. Took your advice and set the velocity before `Timestep` and then set pose.position (and velcocity to zero) after the step
